hackage-security/src/Hackage/Security/TUF/FileInfo.hs

```diff
@@ -87,10 +89,13 @@ fileInfoSHA256 FileInfo{..} = Map.lookup HashFnSHA256 fileInfoHashes
 
 instance Monad m => ToObjectKey m HashFn where
   toObjectKey HashFnSHA256 = return "sha256"
+  toObjectKey HashFnMD5    = return "md5"
+  toObjectKey (HashFnUnknown s) = return s
 
 instance ReportSchemaErrors m => FromObjectKey m HashFn where
   fromObjectKey "sha256" = return HashFnSHA256
-  fromObjectKey str      = expected "valid hash function" (Just str)
```

Comment:
NB: this line is what would currently cause the instance for `fileInfoHashes :: Map HashFn Hash` to choke on unknown keys, if `package.json` files would start to provide an `md5` hash as well...